ls1intum/Orion
# `Feature`: Enable complaints and exams

### Motivation and Context
Orion's half of https://github.com/ls1intum/Artemis/pull/4279. Enables the assessment of complaints, more feedback requests and multiple correction rounds.

### Description
Since the integrated browser JCEF does not support url query params, rewirtes the loading of URLs to now use javascript instead of the built-in methods.

Extends the exercises data with exam related information to insert into the url.

I also found and fixed a few minor code issues like unused imports and deprecated APIs

### Steps for Testing
0. Deploy the linked Artemis PR

1.  Install the release as described in [the readme](https://github.com/ls1intum/Orion/blob/master/README.md#testing-of-pull-requests)
2.  Create/Locate and exercise with open complaints and more feedback requests
3.  Assess them in Orion
    1. Previous feedback should load correctly
    2. The feedback should be editable (for complaints)
    3. The response should be displayed correctly to the student
    4. The back button/restarting should connect with the correct page
4. Create/Locate and exam with a programming exercise and submissions and two correction rounds
5. Assess them in Orion
    1. Both correction rounds should work correctly
    2. Same properties as for complaints should work
